main.go

```diff
@@ -90,9 +90,14 @@ var (
 
 func getApiSpec(apiID string) *APISpec {
 	apisMu.RLock()
-	spec := apisByID[apiID]
-	apisMu.RUnlock()
-	return spec
+	defer apisMu.RUnlock()
+	return apisByID[apiID]
+}
+
+func getApiSpecLen() int {
```

Comment:
I think this name is a tad confusing - you're getting the length of the api list (map), not the length of one api spec. Perhaps `getNumApis`, or `apiSpecsLen`?

Reply:
ok, I will rename it